File: src/clients/ts/src/models/GameView.ts
import { Score } from "./Score";
import { PlayerId } from "./typed-ids";
import { WatchView } from "./WatchView";


export interface GameView {
  initialServer?: PlayerId;
  initialReceiver?: PlayerId;
  currentServer?: PlayerId;
  currentReceiver?: PlayerId;
  points: Score;
  watches: WatchView[];
}

File: src/server/T3.Web/Services/Commit/GameView.cs
using T3.Web.Services.Players;

namespace T3.Web.Services.Commit;

public record GameView
{
    public required PlayerId? InitialServer { get; init; }
    public required PlayerId? InitialReceiver { get; init; }

    public required PlayerId? CurrentServer { get; init; }
    public required PlayerId? CurrentReceiver { get; init; }

    public required Score Points { get; init; } = Score.CreateZero();
    
    /// <summary>
    /// The watches in the current game. They are not part of any set.
    /// </summary>
    public required WatchView[] Watches { get; init; }
}